pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/kit/KitDaoImpl.java

```diff
@@ -554,16 +554,16 @@ public Optional<ScanError> insertKitTracking(KitRequestShipping kitRequestShippi
                 int rowsAffected = stmt.executeUpdate();
                 if (rowsAffected != 1) {
                     dbVals.resultValue = new ScanError(kitRequestShipping.getKitLabel(),
-                            "Kit Label \"" + kitRequestShipping.getKitLabel() + "\" does not exist.\n"
-                                    + UserErrorMessages.IF_QUESTIONS_CONTACT_DEVELOPER);
-                    logger.error("The number of affected rows for kit label " + kitRequestShipping.getKitLabel() + "is not 1.");
+                            String.format("Kit Label \"%s\" does not exist.\n"
+                                    + UserErrorMessages.IF_QUESTIONS_CONTACT_DEVELOPER, kitRequestShipping.getKitLabel()));
+                    logger.error(String.format("The number of affected rows for kit label %s is not 1", kitRequestShipping.getKitLabel()));
```

Comment:
Why doesn't this method rethrow these exceptions? Is it called during message processing? If so, perhaps the exceptions should be handled at the top of the message handling stack?

Reply:
Correct, we can maybe discuss this more and make a plan for refactoring kit-related classes to do better checks?